ui/src/turing/components/form/standard_ensembler/LinkedRoutesTable.js

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect, useMemo, useState } from "react";
+
+import {
+  EuiFlexItem,
+  EuiLoadingChart,
+  EuiTextAlign,
+  EuiInMemoryTable,
+  EuiIcon,
+  EuiTextColor,
+} from "@elastic/eui";
+
+import { useXpApi } from "hooks/useXpApi";
+import moment from "moment";
+import { useConfig } from "config";
+import { getExperimentStatus } from "services/experiment/ExperimentStatus";
+import { LinkedExperimentsContextMenu } from "./LinkedExperimentsContextMenu";
+
+export const LinkedRoutesTable = ({
+  projectId,
+  routes,
+  treatmentConfigRouteNamePath,
+}) => {
+  const { appConfig } = useConfig();
+
+  const [isButtonPopoverOpen, setIsButtonPopoverOpen] = useState(routes.reduce((m, r) => {m[r.id] = {running: false, scheduled: false}; return m}, {}));
+  const [isAllExperimentsLoaded, setIsAllExperimentsLoaded] = useState(false);
+  const [pageIndex, setPageIndex] = useState(0);
+  const [allExperiments, setAllExperiments] = useState([]);
+  const [routeToExperimentMappings, setRouteToExperimentMappings] = useState(routes.reduce((m, r) => {m[r.id] = {running: {}, scheduled: {}}; return m}, {}));
+
+  const { start_time, end_time } = useMemo(
+    () => {
+      let current_time = moment.utc();
+      let start_time = current_time.format(appConfig.datetime.format);
+      let end_time = current_time.add(1000, "y").format(appConfig.datetime.format);
+      return { start_time, end_time };
+      },
+    [appConfig]
+  );
+
+  const [{ data: { data: experiments, paging }, isLoaded }] = useXpApi(
+    `/projects/${projectId}/experiments`,
+    {
+      query: {
+        start_time: start_time,
+        end_time: end_time,
+        page: pageIndex + 1,
```

Comment:
Shall we abstract this into a context function under `ui/src/providers/experiments`, say - `getAllExperiments(status)`? It will help keep this file more streamlined.

Reply:
While we're on this, should we standardise the naming of the context providers we have? Should they be singular or plural? There's currently a mix of both and I'm not if it was intentional to have them that way e.g. we have a `SegmenterContextProvider` in a `segmenters` folder, but a `TreatmentsContextProvider` in a `treatment` folder 👀

Reply:
Haha.. would be nice to standardarise. Since singular is more commonly used, let's use that.

Reply:
Wait.. why was this resolved? Are we not moving this logic to get all experiments into a context function?

Reply:
I think you're viewing an outdated commit :o

Reply:
:O :O my bad.